Flaky test: TestStopContainerWithTimeout (integration/container)

### Description

- related: PR to update this test: https://github.com/moby/moby/pull/45365


Seen failing a few times;

```console
=== CONT  TestStopContainerWithTimeout/3
=== CONT  TestStopContainerWithTimeout/1
    stop_linux_test.go:70: assertion failed: 2 (inspect.State.ExitCode int) != 42 (tc.expectedExitCode int)
--- FAIL: TestStopContainerWithTimeout (0.02s)
    --- PASS: TestStopContainerWithTimeout/0 (0.48s)
    --- FAIL: TestStopContainerWithTimeout/-1 (0.51s)
    --- PASS: TestStopContainerWithTimeout/1 (1.45s)
    --- PASS: TestStopContainerWithTimeout/3 (2.31s)
```

Still flaky 😢 

Based on: 6a0f71cac93c5d255043c91b3443f9db88580e1e

https://github.com/moby/moby/actions/runs/14033473415/job/39286205785?pr=49678#step:8:1422
```
=== Failed
=== FAIL: amd64.integration.container TestStopContainerWithTimeout/big_enough_timeout:_expect_graceful_container_stop (0.24s)
    stop_test.go:108: assertion failed: 2 (inspect.State.ExitCode int) != 42 (tc.expectedExitCode int)
    --- FAIL: TestStopContainerWithTimeout/big_enough_timeout:_expect_graceful_container_stop (0.24s)

=== FAIL: amd64.integration.container TestStopContainerWithTimeout (0.02s)
```

Looks to be quite flaky on rootless indeed; just had it fail twice at least on 2 separate PRs;


```
=== Failed
=== FAIL: amd64.integration.container TestStopContainerWithTimeout/big_enough_timeout:_expect_graceful_container_stop (0.23s)
    stop_test.go:108: assertion failed: 2 (inspect.State.ExitCode int) != 42 (tc.expectedExitCode int)
    --- FAIL: TestStopContainerWithTimeout/big_enough_timeout:_expect_graceful_container_stop (0.23s)

=== FAIL: amd64.integration.container TestStopContainerWithTimeout (0.07s)
```

--- a/integration/container/stop_test.go
+++ b/integration/container/stop_test.go
@@ -67,7 +67,7 @@ func TestStopContainerWithTimeout(t *testing.T) {
 		forcefulKillExitCode = 0x40010004
 	}
 
-	testCmd := container.WithCmd("sh", "-c", "sleep 10 && exit 42")
+	testCmd := container.WithCmd("sh", "-c", "echo ready; sleep 10; exit 42")
 	testData := []struct {
 		doc              string
 		timeout          int
@@ -107,6 +107,7 @@ func TestStopContainerWithTimeout(t *testing.T) {
 			// TODO(vvoland): Investigate why it helps
 			// t.Parallel()
 			id := container.Run(ctx, t, apiClient, testCmd)
+			poll.WaitOn(t, logsContains(ctx, apiClient, id, "ready"))
 
 			_, err := apiClient.ContainerStop(ctx, id, client.ContainerStopOptions{Timeout: &tc.timeout})
 			assert.NilError(t, err)

